src/api/chat.ts

```diff
@@ -80,3 +81,56 @@ export const deletePrivateMessage = async ({
     .collection("chats")
     .doc(messageId)
     .update({ deleted: true });
+
+export const createVideoChat = async (
+  hostId: string,
+  guestId: string,
+  venueId: string
+) => {
+  const videoChatRequest: VideoChatRequest = {
+    hostId: hostId,
+    guestId: guestId,
+    state: VideoChatRequestState.Asked,
+    createdAt: Date.now(),
+  };
+
+  return await firebase
+    .firestore()
+    .collection("experiences")
+    .doc(venueId)
+    .collection("chatRequests")
+    .add(videoChatRequest);
```

Comment:
@0xdevalias @mike-lvov 

Follow up after: https://github.com/sparkletown/sparkle/pull/1385#issuecomment-840691927

My idea is to create the requests like this:
```tsx
 const videoChatRequest: VideoChatRequest = {
    hostId: hostId,
    guestId: guestId,
    state: VideoChatRequestState.Asked,
    createdAt: Date.now(),
  };

await firebase
    .firestore()
    .collection("videoRooms")
    .add(videoChatRequest);
```

And then fetch and store them like this:

```tsx
useFirestoreConnect([
    {
      collection: "videoRooms",
      where: [["guestId", "==", "userId"]]
      storeAs: "videoRooms",
    },
  ]);
```

Reply:
Firstly, see my comments in https://github.com/sparkletown/sparkle/pull/1385#issuecomment-841033488

Reply:
> And then fetch and store them like this

Should that be filtering based on date/similar? Presumably we are only going to care about `videoChatRequests` in a certain state, and potentially only after a certain time (eg. now)?

Also, presumably `where: [["guestId", "==", "userId"]]` would be `where: [["guestId", "==", userId]]` or similar yeah?

Are we only needing to retrieve entries for the `guestId`? What about for the `hostId`?

Reply:
It might make sense to move this out of `api/chat` into something like `api/video` or `api/videoRoom` or similar too.